src/main/java/com/google/gcloud/examples/StorageExample.java

```diff
@@ -336,6 +336,45 @@ public String params() {
     }
   }
 
+  private static class UpdateMetadata extends StorageAction<Tuple<Blob, Map<String, String>>> {
```

Comment:
It's not obvious to me what this is doing. As this is example code lots of comments would help. Refactoring the bits that are not storage related (such as parsing) from the bits that interact with the API would also make it clearer. Using something more specific than Tuple<Blob, Map<String, String>> would also help.

The commands here are objects that can have fields - passing the StorageService in a constructor would separate that, then you could also use bean properties for bucket, path and metadata.